core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -266,8 +266,21 @@ private[spark] class TaskSchedulerImpl(
               assert(availableCpus(i) >= 0)
               launchedTask = true
             }
+          } catch {
+            case e: TaskNotSerializableException => {
+              logError(s"Resource offer failed, task set ${taskSet.name} was not serializable")
```

Comment:
Do we expect the exception to contain any useful information? It might be good to `logError(..., e)`